python/cudf/cudf/core/column/column.py

```diff
@@ -502,7 +502,8 @@ def _wrap_binop_normalization(self, other):
         if other is NA or other is None:
             return cudf.Scalar(other, dtype=self.dtype)
         if isinstance(other, np.ndarray) and other.ndim == 0:
-            other = other.item()
+            # Try and maintain the dtype
+            other = other.dtype.type(other.item())
```

Comment:
There may be awkward corner cases for object arrays here, so it might make sense to say `and other.dtype != object` also (but I can totally support to pragmatically ignore those corner cases here).

Reply:
I think it works:
```
import numpy as np
x = np.asarray(1, dtype=object)
x.dtype.type(x.item()) # => 1 (with type => int)
```